control-plane/p4RuntimeSerializer.cpp

```diff
@@ -1581,6 +1581,7 @@ void P4RuntimeSerializer::serializeP4RuntimeIfRequired(const IR::P4Program *prog
 
 void P4RuntimeSerializer::serializeP4RuntimeIfRequired(const P4RuntimeAPI &p4Runtime,
                                                        const CompilerOptions &options) {
+    // FIXME: get rid of cstring here
```

Comment:
What happens if you replace it with `std::filesystem::path`?

Reply:
There are more dependencies here including `parseFileNames` logic, etc. I decided to leave it as-is for now.